src/Platforms/MySQL/CollationMetadataProvider.php

```diff
@@ -0,0 +1,13 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\DBAL\Platforms\MySQL;
+
+/**
+ * @internal
+ */
+interface CollationMetadataProvider
+{
+    public function getCollationCharset(string $collation): ?string;
```

Comment:
What does `null` mean here? "This collation has no charset"? "We don't know about this collation"? Both? If both, should the second case result in an exception?

Reply:
> What does `null` mean here?

"This collation has no charset" is impossible, so it's "We don't know about this collation".

> […] should the second case result in an exception?

It doesn't have to. Should `strpos()` throw an exception if the substring is not found? The calling code handles `null` by not updating options, so if the collation is invalid, the DDL will eventually fail once it hits the database.

If we decide to throw an exception here, how do we unit-test the comparator without the database connection? Right now, we supply it with a provider that always returns `null`, which is fine because no unit tests rely on this feature.

Reply:
It's clearer, thanks for the explanation.